Draft proposal for a reference deployments working group

emalm opened the pull request.

LeePorte commented:
Looks good to me :+1:

rkoster commented:
What about ensuring updates don't cause downtime. Found for example https://github.com/cloudfoundry/uptimer. This feels like something this working groups could also be responsible for.

loewenstein reviewed (approved):
Looks good. We should remove the "wip" from the pr's title.

emalm commented:
Thanks, @rkoster, added Uptimer to the proposed list of repositories for this WG.

@loewenstein Removed the "[WIP]" from the commit short message and this PR title.

I also added Quarks and bosh-bootloader as proposed scope for this WG, to match the overall structure proposed in #140.

@cloudfoundry/toc For us to proceed, we still need to identify at least the initial set of leads for this WG. Let's solicit interest on this PR and continue discussion during upcoming TOC meetings.

emalm commented:
@Birdrock @davewalter: the TOC would appreciate your help in:
* evaluating the goals and scope of this WG,
* identifying others in the community that would be good candidates to be approvers/leads in this WG,
* listing other technical assets from Release Integration that this WG would own and how the WG might partition them into subareas of responsibility.

@jandubois @manno: same goes, but with a focus on the KubeCF and Quarks scope that would go into this WG.

Thanks!

davewalter commented:
👍 Looks good to me.

➕ The other assets that I think should be here are:
* https://github.com/cloudfoundry/cf-deployment-concourse-tasks/
* https://github.com/cloudfoundry/runtime-ci/
* https://github.com/cloudfoundry/jumpbox-deployment/

ℹ️ The CI tasks in the `runtime-ci` repository are used across projects managed by Release Integration. I think that the majority of them are used by repositories that are being considered for management by this working group, but there may be work needed to split off anything that is unrelated.

ℹ️ Jumpbox-deployment is a component of BOSH Bootloader.

🤔 Did you consider putting BOSH Bootloader in the `Foundational Infrastructure` working group alongside BOSH itself, instead of here in `App Runtime Deployments`? I know that BBL does have some CF knowledge (the ability to provision infrastructure such as load balancers), but it also has similar knowledge for Concourse.

Birdrock commented:
There are quite a few repositories that are cf-for-k8s component repositories. Should those be folded under this umbrella or elsewhere?

emalm commented:
@davewalter Yeah, I think BBL is right on the bubble, so to speak. I tried to transfer it the BOSH PMC some time last year when we were doing clean-up in the Runtime PMC, but the discussion never really got started.

Given that we're likely to get the Deployments WG up and running before the FI WG, I'm going to move BBL to FI for the time being to keep the Deployments WG scope smaller. Then we can evaluate again as we firm up FI and make a final decision about which WG takes it on.

In terms of activity and maintenance, my impression is that BBL has been stable in terms of features for quite a while now but needs periodic bumps of the bosh-deployment and jumpbox-deployment assets that it vendors. Is that correct?

davewalter commented:
> In terms of activity and maintenance, my impression is that BBL has been stable in terms of features for quite a while now but needs periodic bumps of the bosh-deployment and jumpbox-deployment assets that it vendors. Is that correct?

✅ Yes, that is right. There are two fully automated CI pipelines. One bumps the stemcell and CPI releases in `jumpbox-deployment`, and the other bumps both deployments in BBL, sanity tests them against GCP, and cuts a patch release. This is the extent of the legacy process from the old Infrastructure team that I resurrected when I assumed maintenance of BBL a year ago.

There is going to be upcoming work to address issues such as an outdated version of terraform (currently v0.11.15), but that can wait for a final decision to be made about which working group assumes ownership.

emalm commented:
@Birdrock If those repos are primarily to support cf-for-k8s as an integrated reference deployment, I think they belong here. If they're potentially useful independently, they might make sense in one of the other App Runtime WGs, though. Could you suggest a couple example repos for us to reason over?

emalm commented:
@andrew-edgar would also be a great approver or even lead for this WG, with his recent contributions to KubeCF and Quarks and his extensive past history with other areas of CF.

philippthun commented:
Would [cf-performance-tests](https://github.com/cloudfoundry-incubator/cf-performance-tests) be an asset for this WG or does it better fit into the App Runtime Interfaces WG?

rkoster commented:
Any updates on this PR?

stephanme commented:
Not yet but you can expect an update soon.

stephanme commented:
I propose Jochen Ehret (@jochenehret) as working group lead.

Jochen has worked since more than 5 years with Cloud Foundry. He works in a team that consumes cf-deployment for large-scale productive Cloud Foundry deployments at SAP on basically all major IaaS providers. Jochen has in-depth knowledge in several CF components, Bosh and Concourse. He contributed to [cloud_controller_ng](https://github.com/cloudfoundry/cloud_controller_ng) and [cf-performance-tests](https://github.com/cloudfoundry-incubator/cf-performance-tests).

stephanme commented:
I propose @philippthun, @johha, @svkrieger and @iaftab-alam as approvers for this working group even though they have not made direct contributions to cf-deployment. They all have long term experience with cf-deployment, Bosh and Concourse.

@philippthun is an approver for [cloud_controller_ng](https://github.com/cloudfoundry/cloud_controller_ng).

@johha, @svkrieger and @iaftab-alam have contributed to [cloud_controller_ng](https://github.com/cloudfoundry/cloud_controller_ng), [capi-release](https://github.com/cloudfoundry/capi-release) and [cf-performance-tests](https://github.com/cloudfoundry-incubator/cf-performance-tests).

jimconner commented:
Hi Folks - We'd like to volunteer some of the people at Fidelity to help with this in whatever way we can. We're an 8-person team with plenty of concourse, bosh and cf-deployment experience  (ESXi on-prem and AWS). We use concourse for pretty much everything we do in our team and we'd love to assist with building out concourse pipelines for new building/testing new stemcells, validating CF in different envs, or whatever else needs to get done.

Here's a few starter names to throw into the mix - our whole team says they'd like to get involved if they can.

@jimconner, @andrewdriver123, @combor and @shaun7pan

stephanme commented:
Input for the TOC meeting:
- I would love to see todays committers from vmware on the approver list, e.g. @davewalter :-)
- I welcome @jimconner and team to participate in the WG right from the beginning.
- I added some comments on the open questions.

stephanme commented on toc/working-groups/app-runtime-deployments.md:
```diff
@@ -0,0 +1,66 @@
+
+## Open Questions (to be considered before WG formation)
+
+- How much API and behavioral consistency should there be between official CF deployments? - becomes relevant only if a k8s based CF deployment is part of technical assets
```
I guess we can remove this question for now. k8s based CF deployment is out-of-scope for now.

stephanme commented on toc/working-groups/app-runtime-deployments.md:
```diff
@@ -0,0 +1,66 @@
+## Open Questions (to be considered before WG formation)
+
+- How much API and behavioral consistency should there be between official CF deployments? - becomes relevant only if a k8s based CF deployment is part of technical assets
+- How much of that consistency should be the responsibility of this working group, when that consistency may lie within the domain of another working group (for example, one that owns Buildpacks)?
```
My proposal: the consistency requirements should be codified in CATS. CATS should allow an evolution of e.g. buildpacks even if this means incompatibilities (e.g. have different sets of sample apps for old and new buildpacks, controlled by feature flags).

jimconner replied:
I agree wholeheartedly with you on this point. A successful run of CATS is effectively certifying that the deployed environment meets all of the requirements to be able to call itself 'CF'. If there are multiple ways to be a valid 'CF' then CATS should encompass all of them.

stephanme replied:
Added collaboration with other WGs on test suites explicitly to the scope section.

stephanme commented:
Re-requested reviews as they were outdated.

emalm commented on toc/working-groups/app-runtime-deployments.md:
```diff
@@ -0,0 +1,64 @@
+- @andrewdriver123
+- @combor
+- @shaun7pan
+
```
```suggestion
- @ctlong 

```
Request that we add @ctlong from VMware, as he has also been contributing regularly to cf-deployment and CATs.

ctlong replied:
@tjvman has also been contributing regularly to CF-D and CATs. Would it be possible to add him as well?

stephanme replied:
Welcome @ctlong and @tjvman in app runtime deployments WG.

emalm reviewed (commented):
I can only comment, since I'm the PR owner, but LGTM.
